From c641af500c491e1d25c61b592d24a4bf818b4d22 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Fri, 24 Aug 2018 11:51:35 +0200
Subject: [PATCH 49/80] SystemUI: Allow overlaying max notification icons

* I believe there are devices in this world that can fit
  more than 4 notification icons.

Change-Id: I8f01b4e0385b35f4ed93aab2619839e5b40ee0c7
---
 .../SystemUI/res/values/lineage_config.xml    | 20 +++++++++++++++++++
 .../phone/NotificationIconContainer.java      |  6 ++++--
 2 files changed, 24 insertions(+), 2 deletions(-)
 create mode 100644 packages/SystemUI/res/values/lineage_config.xml

diff --git a/packages/SystemUI/res/values/lineage_config.xml b/packages/SystemUI/res/values/lineage_config.xml
new file mode 100644
index 00000000000..73b2589bda1
--- /dev/null
+++ b/packages/SystemUI/res/values/lineage_config.xml
@@ -0,0 +1,20 @@
+<!--
+     Copyright (C) 2018 The LineageOS Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<resources>
+    <!-- Max visible notification icons -->
+    <integer name="config_maxVisibleNotificationIcons">4</integer>
+    <integer name="config_maxVisibleNotificationIconsWhenDark">5</integer>
+</resources>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationIconContainer.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationIconContainer.java
index 5dee2a6000c..e0c735e674c 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationIconContainer.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationIconContainer.java
@@ -127,8 +127,10 @@ public class NotificationIconContainer extends AlphaOptimizedFrameLayout {
         }
     }.setDuration(CONTENT_FADE_DURATION);
 
-    public static final int MAX_VISIBLE_ICONS_WHEN_DARK = 5;
-    public static final int MAX_STATIC_ICONS = 4;
+    public final int MAX_VISIBLE_ICONS_WHEN_DARK =
+            getResources().getInteger(R.integer.config_maxVisibleNotificationIconsWhenDark);
+    public final int MAX_STATIC_ICONS =
+            getResources().getInteger(R.integer.config_maxVisibleNotificationIcons);
     private static final int MAX_DOTS = 1;
 
     private boolean mIsStaticLayout = true;
-- 
2.17.1

